### ISSUE TITLE
A main entrance does not stand out well if the building is a public building.

### ISSUE THREAD

**Issue by geozeisig** [gravitystorm/openstreetmap-carto#3071]
A main entrance does not stand out well if the building is a church, train station or airport passenger building. Example: [Berliner Dom](https://www.openstreetmap.org/node/1967329593), [S-Bahrenfeld](https://www.openstreetmap.org/node/3527774968)
Maybe we should make the entrance **in this** case even darker.

**Comment by kocio-pl** [comment]
@tpikonen What's your take on this? I'm also unsure about #3038.

**Comment by dieterdreist** [comment]
2018-02-14 14:46 GMT+01:00 geozeisig <notifications@github.com>:

> A main entrance does not stand out well if the building is a church, train
> station or airport passenger building. Example: Berliner Dom
> <https://www.openstreetmap.org/node/1967329593>, S-Bahrenfeld
> <https://www.openstreetmap.org/node/3527774968>
> Maybe we should make the entrance *in this* case even darker.
>



the Berliner Dom case is not mapped well, because the entrance isn't part
of the building but only of a building part which is inside the building.

**Comment by tpikonen** [comment]
I agree that these examples look pretty bad. Also, buildings-major and places of worship are also the places where map users probably benefit the most from seeing main entrances.

Technically this can be fixed in the SQL level by getting suitable building/way tags from a spatial intersection of the entrance with the building (similar to the solution proposed in PR #2932) and adjusting the color.

The more straightforward solution would be to make all main entrances dark enough to be visible also in these dark buildings.

**Comment by dieterdreist** [comment]
2018-02-14 16:42 GMT+01:00 tpikonen <notifications@github.com>:

> The more straightforward solution would be to make all main entrances dark
> enough to be visible also in these dark buildings.



or the opposite: very bright

**Comment by polarbearing** [comment]
Could we change the title of this issue, s/public/major/ ?

**Comment by Tomasz-W** [comment]
I've missed a discussion about entrances rendering, and now I find these boxes not intuitive at all. Wouldn't be the classic door symbol from architectural plans suit better here?

https://www.google.pl/search?client=opera&hs=rev&biw=1920&bih=981&tbm=isch&sa=1&ei=mDCFWounD4aVsAGZvIPICQ&q=door+archi+symbol&oq=door+archi+symbol&gs_l=psy-ab.3...20463.21680.0.21985.5.5.0.0.0.0.0.0..0.0....0...1c.1.64.psy-ab..5.0.0....0.JJp_pckK8TA#imgrc=Fwwb35op2dNOKM:

**Comment by dieterdreist** [comment]
sent from a phone

> On 15. Feb 2018, at 12:46, Tomasz Wójcik <notifications@github.com> wrote:
> 
> Wouldn't be the classic door symbol from architectural plans suit better here?



it is not usable for the scales with which we typically deal in OSM and it implies a lot of details (number of parts, direction, opening side and size) which we don’t tag. For entrances the classical symbol is an arrow.

**Comment by tpikonen** [comment]
> I've missed a discussion about entrances rendering, and now I find these boxes not intuitive at all. Wouldn't be the classic door symbol from architectural plans suit better here?

I also thought about this, but this symbol would only work when rotated correctly and getting the direction of the building way requires a rather complicated SQL query. dieterdreist's arguments above against it are also compelling.

I would like to fix this bug and it looks like it requires changes at the SQL level anyway, so maybe the rotated entrance markers will happen. I'm afraid I don't have the time to work on this soon though.

**Comment by jeisenbe** [comment]
Was this issue fixed by #3659 (Lighten major buildings fill and outline) or is it still a problem?

**Comment by pnorman** [comment]
There's almost zero contrast between the entrance square and the major building fill.

![image](https://user-images.githubusercontent.com/1190866/178134543-598656d9-63b1-4a41-964d-13f59aae0be5.png)

For comparison, 
![image](https://user-images.githubusercontent.com/1190866/178134551-629c496f-53d5-4b52-bace-add8f516ed4c.png)

The Dom example may have tagging issues, but it does show the lack of contrast very well
![image](https://user-images.githubusercontent.com/1190866/178134571-8d49f2ed-4c05-4be5-975a-35d92ced74e4.png)

There's two options. One is to change the colour of entrances globally, and the other is to change it for only major buildings. The former is easy, the latter would require querying for buildings the entrance touches. This is practical - the entrances SQL is simple, and we do similar things with turning circles.

**Comment by imagico** [comment]
The other obvious design option would be to change the entrance symbol otherwise to work better on different backgrounds, like for example adding a halo.  Intuitively i'd say neither this or a stronger color would work well - but i can't say for sure without testing.

Keep in mind that the whole concept of so called major buildings is still an open issue (#2532) - it could be advisable to think these two together.

**Comment by dieterdreist** [comment]
current symbol works for me, the access ways are very clear to understand what this is, I’d keep the entrances without white fill, halo or similar, in a deemphasized styling
